### PR TITLE
Impliment UI main page (only UI)

### PR DESCRIPTION
**Pull request by WalterGoldenOwl**
Description Pull Request:
## Todo
- Code UI Main Screen and Tabs inside

## Task
Link of task: https://trello.com/c/p318c6xO

## Screenshot
Screenshot_1696489393
Screenshot_1696489396
Screenshot_1696489398
Screenshot_1696489401

## Video

https://github.com/dustin-goldenowl/jobspot/assets/144195318/a026b79b-bd1f-4b8d-a5b5-aac61ef776fd

### CODE REVIEW

File: lib/src/presentations/connection/widgets/post_item.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        children: [
+          const SizedBox(height: 25),
+          Padding(
+            padding:
+                const EdgeInsets.symmetric(horizontal: AppDimens.smallPadding),
+            child: Column(
+              children: [
+                _buildHeaderPost(
+                  context,
+                  name: "Arnold Leonardo",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/connection/widgets/post_item.dart]: If this's text test, leave a comment this place. If not, should put in app_en.arb

File: lib/src/presentations/connection/widgets/post_item.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                  time: "21 minuts ago",
+                ),
+                const SizedBox(height: 20),
+                Text(
+                  "What are the characteristics of a fake job call form?",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/connection/widgets/post_item.dart]: Using string resource app_en.arb

File: lib/src/presentations/connection/widgets/post_item.dart [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      decoration: BoxDecoration(
+        color: const Color(0xFF3F13E4).withOpacity(0.1),
+        borderRadius: const BorderRadius.vertical(bottom: Radius.circular(20)),
+      ),
+      height: 64,
+      padding: const EdgeInsets.symmetric(horizontal: AppDimens.mediumPadding),
+      child: Row(
+        children: [
+          _buildItemReaction(
+            onTap: () {},
```
Comment (dustin-goldenowl) [inline lib/src/presentations/connection/widgets/post_item.dart]: If this function is implement later, you should leave a comment `// TODO: description to action`

File: lib/src/presentations/connection/widgets/post_item.dart [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  }) {
+    return GestureDetector(
+      onTap: onTap,
+      child: Row(
+        children: [
+          icon,
+          const SizedBox(width: 5),
+          Text(
+            "$quantity",
+            style: const TextStyle(color: Color(0xFF6F6B80)),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/connection/widgets/post_item.dart]: Using Style from define class and color from resource define

File: lib/src/presentations/home/screen/home_page.dart [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+              width,
+              onTap: () {},
+            ),
+            const SizedBox(width: 20),
+            Column(
+              children: [
+                _buildJobContainer(
+                  width: width,
+                  title: AppLocal.text.home_page_full_time,
+                  quantity: "66.8k",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/home/screen/home_page.dart]: Remove hard code

File: lib/src/presentations/home/screen/home_page.dart [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+                  color: const Color(0xFFBEAFFE),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/home/screen/home_page.dart:142]: Define color

File: lib/src/presentations/home/screen/home_page.dart [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+                  onTap: () {},
+                ),
+                const SizedBox(height: 20),
+                _buildJobContainer(
+                  width: width,
+                  title: AppLocal.text.home_page_part_time,
+                  quantity: "38.9k",
+                  color: const Color(0xFFFFD6AD),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/home/screen/home_page.dart:150]: The same above

File: lib/src/presentations/main/screen/main_view.dart [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+  void showBottomSheet(BuildContext context) {
+    showModalBottomSheet(
+      context: context,
+      shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.vertical(top: Radius.circular(20)),
+      ),
+      backgroundColor: Colors.white,
+      builder: (context) {
+        return Container(
```
Comment (dustin-goldenowl) [inline lib/src/presentations/main/screen/main_view.dart]: You should separate widget contain of bottom sheet to new widget class.

File: lib/src/presentations/main/screen/main_view.dart [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        itemCount: 4,
+        onPageChanged: (value) => cubit.changeIndex(value),
+        itemBuilder: (context, index) {
+          return PageStorage(bucket: cubit.bucket, child: cubit.screens[index]);
+        },
+      ),
+    );
+  }
+
+  void showBottomSheet(BuildContext context) {
```
Comment (dustin-goldenowl) [inline lib/src/presentations/main/screen/main_view.dart]: should be put in MainCubit

File: lib/src/presentations/notification/widgets/notification_item.dart [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+      onTap: onTap,
+      child: Container(
+        padding: const EdgeInsets.all(AppDimens.smallPadding),
+        decoration: BoxDecoration(
+          color:
+              isRead ? const Color(0xFF7551FF).withOpacity(0.1) : Colors.white,
+          borderRadius: BorderRadius.circular(16),
+          boxShadow: const [
+            BoxShadow(
+              color: Color.fromARGB(18, 153, 171, 198),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/notification/widgets/notification_item.dart]: Define color

File: lib/src/presentations/notification/widgets/notification_item.dart [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: onTap,
+      child: Container(
+        padding: const EdgeInsets.all(AppDimens.smallPadding),
+        decoration: BoxDecoration(
+          color:
+              isRead ? const Color(0xFF7551FF).withOpacity(0.1) : Colors.white,
```
Comment (dustin-goldenowl) [inline lib/src/presentations/notification/widgets/notification_item.dart]: Define color

File: lib/src/presentations/notification/widgets/notification_item.dart [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    "Google",
+                    style: AppStyles.boldTextHaiti.copyWith(fontSize: 16),
+                  ),
+                  const SizedBox(height: 6),
+                  Text(
+                    "Applications for Apple companies have entered for company review",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/notification/widgets/notification_item.dart]: Using app_en.arb

File: lib/src/presentations/notification/widgets/notification_item.dart [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+                    style: AppStyles.normalTextMulledWine,
+                  ),
+                  const SizedBox(height: 15),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      Text(
+                        "1 Day ago",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/notification/widgets/notification_item.dart]: Remove hard code

File: lib/src/presentations/notification/widgets/notification_item.dart [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+                        style: AppStyles.normalTextSpunPearl,
+                      ),
+                      TextButton(
+                        onPressed: () {},
+                        style: TextButton.styleFrom(
+                          foregroundColor: const Color(0xFFFF4B4B),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/notification/widgets/notification_item.dart]: Define color

File: lib/src/presentations/save_job/screen/save_job_page.dart [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+              ),
+              onTap: () {},
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  void showBottomSheetOption(BuildContext context) {
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/screen/save_job_page.dart]: Put in Cubit class

File: lib/src/presentations/save_job/screen/save_job_page.dart [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+    showModalBottomSheet(
+      context: context,
+      shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.vertical(top: Radius.circular(20)),
+      ),
+      backgroundColor: Colors.white,
+      builder: (context) {
+        return Container(
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/screen/save_job_page.dart]: You should separate widget contain of bottom sheet to new widget class.

File: lib/src/presentations/save_job/screen/save_job_page.dart [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+          width: double.infinity,
+          height: 300,
+          child: Column(
+            children: [
+              const SizedBox(height: 25),
+              Container(
+                height: 4,
+                width: 40,
+                decoration: BoxDecoration(
+                  color: const Color(0xFF5B5858),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/screen/save_job_page.dart]: Define color

File: lib/src/presentations/save_job/screen/save_job_page.dart [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+                  borderRadius: BorderRadius.circular(20),
+                ),
+              ),
+              const SizedBox(height: 50),
+              _buildOption(
+                onTap: () {
+                  context.router.pop();
+                  customToast(context, text: "Tính năng chưa ra mắt");
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/screen/save_job_page.dart]: using app_en.arb

File: lib/src/presentations/save_job/screen/save_job_page.dart [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      onTap: onTap,
+      child: Row(
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          SvgPicture.asset(
+            icon,
+            colorFilter: const ColorFilter.mode(
+              Color(0xFF524B6C),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/screen/save_job_page.dart]: The same above

File: lib/src/presentations/save_job/widgets/custom_job_card.dart [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+                    ),
+                  ),
+              ],
+            ),
+            const SizedBox(height: 20),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(
+                  "25 minute ago",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/widgets/custom_job_card.dart]: Hard code

File: lib/src/presentations/save_job/widgets/custom_job_card.dart [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+                  style: AppStyles.normalTextSpunPearl,
+                ),
+                RichText(
+                  text: TextSpan(
+                    children: [
+                      const TextSpan(
+                        text: "\$15K",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/save_job/widgets/custom_job_card.dart]: The same above

File: lib/src/presentations/connection/widgets/post_item.dart (hunk 1 above)
Comment (ngoctienUIT) [reply lib/src/presentations/connection/widgets/post_item.dart:26]: I writed comment

Comment (WalterGoldenOwl) [reply lib/src/presentations/connection/widgets/post_item.dart]: I wrote a comment

File: lib/src/presentations/connection/widgets/post_item.dart (hunk 2 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/connection/widgets/post_item.dart]: This is a hard text to test UI. I wrote a comment.

File: lib/src/presentations/connection/widgets/post_item.dart (hunk 3 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/connection/widgets/post_item.dart]: I wrote a comment TODO

File: lib/src/presentations/home/screen/home_page.dart (hunk 5 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/home/screen/home_page.dart]: I write hard text to test UI. I wrote a comment

File: lib/src/presentations/save_job/widgets/custom_job_card.dart (hunk 20 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/save_job/widgets/custom_job_card.dart]: I wrote a comment

File: lib/src/presentations/connection/widgets/post_item.dart (hunk 4 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/connection/widgets/post_item.dart]: I fixed it

File: lib/src/presentations/notification/widgets/notification_item.dart (hunk 11 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/notification/widgets/notification_item.dart]: I fixed it

File: lib/src/presentations/save_job/screen/save_job_page.dart (hunk 18 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/save_job/screen/save_job_page.dart]: I fixed it

File: lib/src/presentations/notification/widgets/notification_item.dart (hunk 10 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/notification/widgets/notification_item.dart]: I fixed it

File: lib/src/presentations/main/screen/main_view.dart [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+    final cubit = context.read<MainCubit>();
+    return Scaffold(
+      bottomNavigationBar: BlocBuilder<MainCubit, MainState>(
+        buildWhen: (previous, current) =>
+            previous.currentIndex != current.currentIndex,
+        builder: (context, state) {
+          return CustomBottomBar(
+            changeTab: (index) {
+              cubit.changeIndex(index);
+              cubit.toPage(index);
```
Comment (dustin-goldenowl) [inline lib/src/presentations/main/screen/main_view.dart]: Should be put cubit.toPage into cubit.changeIndex.
This place only call cubit.changeIndex

Comment (WalterGoldenOwl) [reply lib/src/presentations/main/screen/main_view.dart]: I fix it

File: lib/src/presentations/main/screen/main_view.dart (hunk 9 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/main/screen/main_view.dart]: I fixed it

File: lib/src/presentations/main/screen/main_view.dart (hunk 8 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/main/screen/main_view.dart]: I fixed it

File: lib/src/presentations/save_job/screen/save_job_page.dart (hunk 15 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/save_job/screen/save_job_page.dart]: I fixed it

File: lib/src/presentations/save_job/screen/save_job_page.dart (hunk 16 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/save_job/screen/save_job_page.dart]: I fixed it

**Pull request merged**